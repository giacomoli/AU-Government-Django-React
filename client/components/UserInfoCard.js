import React, { Component, Fragment } from 'react'
import PropTypes from 'prop-types'
import { gql, graphql } from 'react-apollo'
import swal from 'sweetalert'
import { BACKEND_URL } from 'constants/page'
import Button from 'components/Button'

const defaultAvatar = '/static/assets/images/users/default-avatar.png'

class UserInfo extends Component {
  static propTypes = {
    profile: PropTypes.shape({
      photo: PropTypes.string,
      fullName: PropTypes.string,
      email: PropTypes.string,
    }).isRequired,
  }

  constructor(props) {
    super(props)

    this.state = {
      isSync: false,
    }
  }

  get address() {
    const { location } = this.props.profile
    return `${location.address1 && `${location.address1}, `}${location
      .address2 && `${location.address2}, `} ${location.suburb &&
      `${location.suburb}, `} ${location.stateDisplay &&
      `${location.stateDisplay}, `}${location.postCode &&
      `${location.postCode}`}${location.country &&
      `, ${location.country}`}`
  }

  handleSyncMails = async () => {
    this.setState({ isSync: true })

    const res = await this.props.syncMails()
    const { success, errors } = res.data.syncMails

    if (success) {
      swal({ title: 'Successfully started downloading emails!', text: 'This process can take some time, please check back later', icon: 'success' })
    } else {
      swal({ title: 'Something went wrong!', text: errors.join('\n'), icon: 'error' })
    }

    this.setState({ isSync: false })
  }

  render() {
    const { photo, fullName, gmail, email, mobile } = this.props.profile
    const { isSync } = this.state

    return (
      <div className="card">
        <div className="card-body">
          <center className="m-t-30">
            <img
              src={`${BACKEND_URL}${photo}` || defaultAvatar}
              alt="avatar"
              className="img-circle"
              width="150"
            />
            <h4 className="card-title m-t-10">{fullName}</h4>
            {
              gmail && <Button
                loading={isSync}
                icon="fa fa-inbox"
                className="btn btn-rounded btn-primary"
                title="Sync mails"
                onClick={this.handleSyncMails}
              />
            }
          </center>
          {gmail &&
            <div className="ribbon ribbon-info ribbon-left" style={{ left: -10 }}>
              Gmail activated
            </div>
          }
        </div>
        <div>
          <hr />
        </div>
        <div className="card-body">
          <small className="text-muted">Email address</small>
          <h6>{email}</h6>
          <small className="text-muted p-t-30 db">Phone</small>
          <h6>{mobile}</h6>
          <small className="text-muted p-t-30 db">Address</small>
          <h6>{this.address}</h6>
          {gmail &&
            <Fragment>
              <small className="text-muted p-t-30 db">Gmail</small>
              <h6>{gmail}</h6>
            </Fragment>
          }
        </div>
      </div>
    )
  }
}

const syncMails = gql`
  mutation syncMails {
    syncMails {
      success
      errors
    }
  }
`

export default graphql(syncMails, { name: 'syncMails' })(UserInfo)
