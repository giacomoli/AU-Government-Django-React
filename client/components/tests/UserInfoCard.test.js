import React from 'react'
import renderer from 'react-test-renderer'
import { ApolloProvider } from 'react-apollo';
import { reducer } from 'redux-form'
import { ApolloClient } from 'apollo-client';
import { createStore } from 'redux';
import UserInfoCard from '../UserInfoCard'

/* eslint-disable */
describe('UserInfoCard card', () => {
  const client = new ApolloClient()
  const store = createStore(reducer)
  const profile = {
      id: 'VXNlclR5cGU6NTI=',
      photo: '/image.png',
      fullName: 'Admin Qwe',
      firstName: 'Adin',
      lastName: 'Qwe',
      email: 'qwe@qwe.qwe',
      mobile: '123-12-12',
      location: {
        __typename: "LocationType",
        id: "qweqqwe",
        address1: 'ad1',
        address2: 'ad2',
        stateDisplay: 'SU',
        suburb: '123',
        postCode: '123',
        country: 'co',
      },
  }


  it('should render UserInfoCard', () => {

    const wrapper = renderer.create(
      <ApolloProvider client={client} store={store}>
        <UserInfoCard profile={profile} />
      </ApolloProvider>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
