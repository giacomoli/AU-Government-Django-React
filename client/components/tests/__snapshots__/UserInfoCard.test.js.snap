// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserInfoCard card should render UserInfoCard 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <center
      className="m-t-30"
    >
      <img
        alt="avatar"
        className="img-circle"
        src="http://localhost:8000/image.png"
        width="150"
      />
      <h4
        className="card-title m-t-10"
      >
        Admin Qwe
      </h4>
    </center>
  </div>
  <div>
    <hr />
  </div>
  <div
    className="card-body"
  >
    <small
      className="text-muted"
    >
      Email address
    </small>
    <h6>
      qwe@qwe.qwe
    </h6>
    <small
      className="text-muted p-t-30 db"
    >
      Phone
    </small>
    <h6>
      123-12-12
    </h6>
    <small
      className="text-muted p-t-30 db"
    >
      Address
    </small>
    <h6>
      ad1, ad2,  123,  SU, 123, co
    </h6>
  </div>
</div>
`;
